GradeCenter.API/Controllers/SchoolController.cs

```diff
@@ -9,34 +13,71 @@ namespace GradeCenter.API.Controllers
     public class SchoolController : ControllerBase
     {
         private readonly ISchoolService _schoolService;
+        private readonly IMapper _mapper;
 
-        public SchoolController(ISchoolService schoolService)
+        public SchoolController(ISchoolService schoolService, IMapper mapper)
         {
             _schoolService = schoolService;
+            _mapper = mapper;
+        }
+        /// <summary>
+        /// Reads all the existing School entities in the database.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("Read")]
+        public async Task<IActionResult> Read()
+        {
+            return Ok(_schoolService.Read());
         }
 
         /// <summary>
         /// Creates an object of type School and stores it in the database.
         /// </summary>
         /// <param name="requestModel"></param>
         /// <returns></returns>
+        [Authorize]
         [HttpPost("Create")]
         public async Task<IActionResult> Create(SchoolCreateRequestModel requestModel)
         {
-            await _schoolService.Create(requestModel.Name, requestModel.Address, requestModel.PrincipalId, requestModel.PrincipalFirstName, requestModel.PrincipalLastName);
+            if (this.User.Identity.IsAuthenticated) return Unauthorized();
```

Comment:
Have you tested this? I don't think it should be working. Here you should use "UserManager" instead, like that:

`User loggedUser = (User)await _userManager.FindByNameAsync(User.Identity.Name);`